client/views/room/lib/QuickActions/generator.tsx

```diff
@@ -0,0 +1,42 @@
+import { Emitter, EventHandlerOf } from '@rocket.chat/emitter';
+
+
+export type Store<T> = Map<string, T>;
+
+
+export type Events<T> = {
+	change: Store<T>;
+};
+
+export const generator = function generator<T>(name?: string): ({
+	store: Store<T>;
+	add: (id: string, action: T) => Store<T>;
+	remove: (id: string) => boolean;
+	listen: (handler: EventHandlerOf<Events<T>, 'change'>) => Function;
+	name: string | undefined;
+}) {
+	const store: Store<T> = new Map();
+	const emitter = new Emitter<Events<T>>();
+
+	const add = (id: string, action: T): Store<T> => {
+		store.set(id, action);
+		emitter.emit('change', store);
+		return store;
+	};
+
+	const remove = (id: string): boolean => {
+		const result = store.delete(id);
+		emitter.emit('change', store);
+		return result;
+	};
+
+	const listen = (handler: EventHandlerOf<Events<T>, 'change'>): Function => emitter.on('change', handler);
```

Comment:
is this not the same generator we have been using on other places? why is this code duplicated here?